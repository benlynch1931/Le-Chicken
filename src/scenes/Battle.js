import React, { useContext } from 'react';
import { Image, View } from 'react-native';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';

const Battle = () => {

  return (
    <View>
    </View>
  )
}
export default Battle;
