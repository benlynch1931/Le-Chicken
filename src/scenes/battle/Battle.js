import React, { useContext } from 'react';
import { View } from 'react-native';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';

import BattleContextProvider from '../../contexts/BattleContext.js'
import { GameContext } from '../../contexts/GameContext.js'

import ChickenOpponent from './ChickenOpponent.js'
import BattleChicken from './BattleChicken.js'
import AttackCommands from './AttackCommands.js'
import BattleView from './BattleView.js'
import HealthBar from './HealthBar.js'
import FenceOpponent from './FenceOpponent.js'


const Battle = () => {
  const { level } = useContext(GameContext)

  const opponent = () => {
    // Swapped levels for spiking
    if (level === 5) {
      return (
        <ChickenOpponent></ChickenOpponent>
      )
    } else if (level === 4) {
      return (
        <FenceOpponent></FenceOpponent>
      )
    }
  } 

  return (
    <View>
      <BattleContextProvider>
        <HealthBar character={"Chicken"}/>
        <BattleChicken></BattleChicken>
        { opponent() }
        <HealthBar character={"Opponent"}/>
        <BattleView></BattleView>
        <AttackCommands></AttackCommands>
      </BattleContextProvider>
    </View>
  )

}

export default Battle;
