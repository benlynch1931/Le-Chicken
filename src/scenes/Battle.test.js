import React from 'react';
import { create, act } from 'react-test-renderer';
import Battle from './Battle.js';

const mockContext = {level: 4}

describe('Battle', () => {

  it('render correctly', () => {
    const battle = create(<Battle />)
    expect(battle).toMatchSnapshot();
  })
})
