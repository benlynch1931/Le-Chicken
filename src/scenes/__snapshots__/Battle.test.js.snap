// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Battle render correctly 1`] = `<View />`;

exports[`Battle renders correctly 1`] = `<View />`;
